src/icp_v2.cc

```diff
@@ -680,16 +662,26 @@ icpHandleUdp(int sock, void *)
         debugs(12, 4, "icpHandleUdp: FD " << sock << ": received " <<
                (unsigned long int)len << " bytes from " << from);
 
-#ifdef ICP_PACKET_DUMP
-
-        icpPktDump(buf);
-#endif
-
         if ((size_t) len < sizeof(icp_common_t)) {
             debugs(12, 4, "icpHandleUdp: Ignoring too-small UDP packet");
             break;
         }
 
+        auto pkt = reinterpret_cast<icp_common_t *>(buf);
```

Comment:
This cast exposes Squid to unaligned integer access dangers, right? If Squid already had this bug in icpHandleIcpV2() and similar code, then please add an XXX here (at least). Otherwise, we should not add these dangers, especially for the debugging sake.

Reply:
Maybe. This is a pre-existing if so. Added the XXX with next update.

Reply:
fixed in 920f275